join.spec.md

```diff
@@ -100,204 +101,347 @@ flowchart TB
 <a name=def-composer>**Composers** (or **compilers**)</a> are producers which compose subgraphs into a supergraph. This document places no particular requirements on the composition algorithm, except that it must produce a valid supergraph.
 
 <a namme=def-router>**Routers**</a> are consumers which serve a composed schema as a GraphQL endpoint. *This definition is non-normative.*
-  - Graph routers differ from standard GraphQL endpoints in that they are not expected to process data or communicate with (non-GraphQL) backend services on their own. Instead, graph routers receive GraphQL requests and service them by performing additional GraphQL requests. This spec provides guidance for implementing routers, but does not require particular implementations of query separation or dispatch, nor does it attempt to normatively separate routers from other supergraph consumers.
-  - Routers will often omit schema elements from the schema they present to clients via introspection ({join__Graph}, for example, will typically be omitted)
+  - Graph routers differ from standard GraphQL endpoints in that they are not expected to resolve fields or communicate with (non-GraphQL) backend services on their own. Instead, graph routers receive GraphQL requests and service them by performing additional GraphQL requests. This spec provides guidance for implementing routers, but does not require particular implementations of query separation or dispatch, nor does it attempt to normatively separate routers from other supergraph consumers.
+  - Routers will often omit schema elements from the API schema that is used to validate client operations and that they present to clients via introspection ({join__Graph}, for example, will typically be omitted)
```

Comment:
API schema should probably be defined if we're going to use it here. FWIW, I find the previous iteration plenty expressive and a bit less of a mouthful. Is "exposed via introspection" an incomplete description that this is needed?

It's obviously not wrong, but maybe redundant and the sentence may be getting a bit long to read.

Reply:
I linked API schema to the core spec. 9673ed4

I think it's important to be clear that the schema we're talking about is used for validation. Plenty of people turn off introspection; we want to make sure they understand that this is still a meaningful concept in that case.  (eg, as soon as we implement inaccessible fields this will be quite relevant.)

I agree this could be less of a mouthful while still containing both places that the API schema is used, though...

Reply:
Is a51b5a2a step in the right direction?